src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,153 @@
+            node.prev = newNode;
+            if (left == null) {
+                first = newNode;
+            } else {
+                left.next = newNode;
+            }
+            size++;
+        }
     }
 
     @Override
     public void addAll(List<T> list) {
+        for (T elementList : list) {
+            add(elementList);
+        }
     }
 
     @Override
     public T get(int index) {
-        return null;
+        checkElementIndex(index);
+        return node(index).element;
     }
 
     @Override
     public T set(T value, int index) {
-        return null;
+        checkElementIndex(index);
+        Node<T> current = node(index);
+        T oldVal = current.element;
+        current.element = value;
+        return oldVal;
     }
 
     @Override
     public T remove(int index) {
-        return null;
+        checkElementIndex(index);
+        return unlink(node(index));
     }
 
     @Override
     public boolean remove(T object) {
+        for (Node<T> current = first; current != last.next; current = current.next) {
+            if (current.element == object || object != null && object.equals(current.element)) {
+                unlink(current);
+                return true;
+            }
+        }
         return false;
     }
 
+    T unlink(Node<T> current) {
+        final T element = current.element;
+        final Node<T> next = current.next;
+        final Node<T> prev = current.prev;
+        if (prev == null) {
+            first = next;
+        } else {
+            prev.next = next;
+            current.prev = null;
+        }
+        if (next == null) {
+            last = prev;
+        } else {
+            next.prev = prev;
+            current.next = null;
+        }
+        current.element = null;
+        size--;
+        return element;
+    }
+
     @Override
     public int size() {
-        return 0;
+        return size;
     }
 
     @Override
     public boolean isEmpty() {
-        return false;
+        return size == 0;
+    }
+
+    Node<T> node(int index) {
+        checkElementIndex(index);
+        Node<T> current;
+        if (index < size / 2) {
+            current = first;
+            for (int i = 0; i < index; i++) {
+                current = current.next;
+            }
+        } else {
+            current = last;
+            for (int i = (size - 1); i > index; i--) {
+                current = current.prev;
+            }
+        }
+        return current;
+    }
+
+    private static class Node<T> {
+        private T element;
+        private Node<T> prev;
+        private Node<T> next;
+
+        Node(Node<T> prev, T element, Node<T> next) {
+            this.element = element;
+            this.prev = prev;
+            this.next = next;
+        }
+    }
+
+    private void checkElementIndex(int index) {
+        if (index < 0 || index >= size) {
+            throw new IndexOutOfBoundsException("Can't element index: " + index);
+        }
```

Comment:
You can delete this method and add this logic to your method node(index), because you use those methods only together in your code.